WordPress/src/main/java/org/wordpress/android/ui/bloggingprompts/list/BloggingPromptsListFragment.kt

```diff
@@ -0,0 +1,130 @@
+        binding = BloggingPromptsListFragmentBinding.inflate(inflater)
+        return binding.root
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        val section = arguments?.getSerializable(LIST_TYPE) as? PromptSection
+        val site = parentViewModel.getSite()
+        initializeViews()
+        setupObservers()
+
+        viewModel.onOpen(site, section)
+    }
+
+    private fun initializeViews() {
+        with(binding.promptsList) {
+            layoutManager = LinearLayoutManager(activity, RecyclerView.VERTICAL, false)
+            addItemDecoration(DividerItemDecoration(context, LinearLayout.VERTICAL))
+            promptsListAdapter = BloggingPromptsListAdapter()
+            adapter = promptsListAdapter
+        }
+    }
+
+    private fun setupObservers(){
+        viewModel.contentViewState.observe(viewLifecycleOwner) {
+            setContentViewVisibility(
+                    it.isVisible,
+                    it.list
+            )
+        }
+        viewModel.errorViewState.observe(viewLifecycleOwner) {
+            setErrorViewVisibility(
+                    it.isVisible,
+                    it.imageResId,
+                    it.titleTextResId,
+                    it.subtitleTextResId,
+                    it.buttonTextResId
+            )
+        }
+    }
+
+    private fun setContentViewVisibility(
+        isVisible: Boolean = false,
+        list: List<BloggingPromptsListItem> = emptyList()
+    ) {
+        promptsListAdapter.update(list)
+        binding.promptsList.setVisible(isVisible)
+    }
+
+    private fun setErrorViewVisibility(
+        isVisible: Boolean = false,
+        @DrawableRes imageResId: Int? = null,
+        @StringRes titleTextResId: Int? = null,
+        @StringRes subtitleTextResId: Int? = null,
+        @StringRes buttonTextResId: Int? = null
+    ) {
+        with(binding.actionableEmptyView) {
+            setVisible(isVisible)
+            with(image) {
+                imageResId?.let {
+                    setVisible(true)
+                    setImageResource(it)
+                } ?: setVisible(false)
+            }
+            with(title) {
+                titleTextResId?.let {
+                    setVisible(true)
+                    setText(it)
+                } ?: setVisible(false)
+            }
+            with(subtitle) {
+                subtitleTextResId?.let {
+                    setVisible(true)
+                    setText(it)
+                } ?: setVisible(false)
+            }
+            with(button) {
+                buttonTextResId?.let {
+                    setVisible(true)
+                    setText(it)
+                    setOnClickListener {
+                        viewModel.onClickButtonRetry()
+                    }
+                } ?: setVisible(false)
+            }
+        }
```

Comment:
maybe we could simplify all this section injecting a UiHelpers that has a few utility functions that may help. wdyt?

Reply:
Makes sence. I will take a look and reuse where applicable. 

In fact, I noticed that one of the methods I introduced in an earlier PR `getQuantityString` (to handle plurals) already existed in another utility class. Will remove my implementation in the next Pr.

Reply:
Done.